### ISSUE TITLE
Expose a way to resolve alias-qualified symbol in custom hooks

### ISSUE THREAD

**Issue by camsaul** [clj-kondo/clj-kondo#1996]
It would be great if there was a way to take a symbol/token node using a namespace alias and get a fully-qualified version of a symbol. For example, 

```clj
(some-fn 'str/includes?)
;; => 'clojure.string/includes?
```

In the normal Clojure world you could figure this out yourself by looking `ns-aliases`, so if we had something like that for the namespace being linted that would also be a way to resolve this issue.

Motivating example: I'm working on a custom linter that will complain if you use certain forms e.g. `with-redefs` inside parallel tests (we're using a test runner that supports parallel tests). I have it working where it can walk the node and find the exact symbol `with-redefs` and error, but it doesn't detect situations if you do something like

```clj
(ns my-ns
  (:require [clojure.core :as core]))

(deftest ^:parallel my-test
  (core/with-redefs [...]
    ...))
```

I would like my hook to be able to take a `with-redefs` token node **or** `core/with-redefs` token node and figure out that in this specific namespace, it's referring to `clojure.core/with-redefs`.
